lib/fog/vsphere/requests/compute/get_storage_pod.rb

```diff
@@ -23,6 +23,7 @@ def get_raw_storage_pod(name, datacenter_name)
 
       class Mock
         def get_storage_pod(name, datacenter_name)
+          list_storage_pods().select{|h| h[:name] == name }.first
```

Comment:
I believe the mock version of the api should be more like the real one. Can't you do a search based on the `datacenter_name`?